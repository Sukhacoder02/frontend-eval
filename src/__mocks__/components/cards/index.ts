export const mockCardsProps = {
  theme: 1,
  allClicked: true,
  registeredClicked: false,
  bookMarkedClicked: false,
  seatsClicked: false,
};
