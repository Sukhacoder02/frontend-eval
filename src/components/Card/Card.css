.card {
  width: 350px;
  height: 500px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  color: white;
}
.cardImage {
  width: 100%;
  height: 45%;
  object-fit: cover;
}

.cardImage > img {
  width: 100%;
  height: 100%;
  object-fit: cover;
}
.cardDetails {
  height: 55%;
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: center;
  gap: 7px;
  background-color: #ea8282;
  padding: 5px;
}
.reactions {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  gap: 10px;
  width: 100%;
  box-sizing: border-box;
  padding: 5px;

  background-color: #ea8282;
}
.register {
  width: 50%;
  display: flex;
  flex-direction: row;
  align-items: flex-start;
}
.bookmark {
  display: flex;
  flex-direction: row;
  align-items: flex-end;
  justify-content: flex-end;
  width: 50%;
}
.cardDescription {
  text-align: center;
}
.venue {
  text-align: center;
}
.date {
  text-align: center;
}
