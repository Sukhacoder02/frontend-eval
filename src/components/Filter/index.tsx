import * as React from 'react';
import './Filter.css';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faFilter, faChevronUp, faChevronDown, faMagnifyingGlass } from '@fortawesome/free-solid-svg-icons';
interface FilterProps {
  handleDropDown: () => void;
  dropDown: boolean;
  setSearchQuery: (value: string) => void;
}
const Filter: React.FC<FilterProps> = (props: FilterProps): JSX.Element => {
  const handleChange = (e: any) => {
    props.setSearchQuery(e.target.value);
  };
  return (
    <div className="filter">
      <div className="filterAndSearch">
        <div className="filterIcon">
          <FontAwesomeIcon icon={faFilter} />
          <p>FILTER</p>
          <button className="dropDown" onClick={props.handleDropDown}>
            <FontAwesomeIcon icon={props.dropDown ? faChevronUp : faChevronDown} />
          </button>
        </div>
        <div className="searchBar">
          <input onChange={handleChange} type="text" placeholder="EVENT NAME" />
          <button className="search">
            <FontAwesomeIcon icon={faMagnifyingGlass} />
          </button>
        </div>
      </div>
    </div>
  );
};
export default Filter;
