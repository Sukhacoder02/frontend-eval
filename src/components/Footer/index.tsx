import * as React from 'react';
import { GET_THEMES, PUT_THEME } from '../../constants/apiEndPoints';
import makeRequest from '../../utils/MakeRequest';
import './Footer.css';
import { FooterProps } from '../../types';
const Footer: React.FC<FooterProps> = (props: FooterProps): JSX.Element => {
  const [prefferedTheme, setPrefferedTheme] = React.useState('1');
  React.useEffect(() => {
    makeRequest(GET_THEMES).then(data => {
      setPrefferedTheme(data.preferredThemeId);
    });
  }, []);
  const [selectedTheme, setSelectedTheme] = React.useState('1');

  const handleThemeCheck = (event: any) => {
    setSelectedTheme(event.target.id.substring(5));
  };
  const handleThemeClick = async (event: any) => {
    await makeRequest(PUT_THEME, {
      data: {
        preferredThemeId: Number(selectedTheme),
      },
    });
    props.reload();
  };
  return (
    <footer className={'theme' + props.theme}>
      <div className="wrapFooter">
        <div className="themes">
          <p>THEMES</p>
          <button onClick={handleThemeCheck}>
            <div id="theme1" className="theme"></div>
          </button>
          <button onClick={handleThemeCheck}>
            <div id="theme2" className="theme"></div>
          </button>
          <button onClick={handleThemeCheck}>
            <div id="theme3" className="theme"></div>
          </button>
        </div>
        <div className="button">
          <button className="save" onClick={handleThemeClick}>
            SAVE THEME
          </button>
        </div>
      </div>
    </footer>
  );
};
export default Footer;
