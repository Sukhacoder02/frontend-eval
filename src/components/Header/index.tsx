import './Header.css';
import * as React from 'react';
import { HeaderProps } from '../../types';
const Header: React.FC<HeaderProps> = (props: HeaderProps): JSX.Element => {
  return (
    <nav className={'theme' + props.theme}>
      <div className="navbar">
        <h1>EVENTIFY</h1>
      </div>
    </nav>
  );
};
export default Header;
