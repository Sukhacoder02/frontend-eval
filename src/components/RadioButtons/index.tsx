import * as React from 'react';
import './RadioButton.css';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCircle, faCircleDot } from '@fortawesome/free-solid-svg-icons';

import { RadioButtonProps } from '../../types';

const RadioButtons: React.FC<RadioButtonProps> = (props: RadioButtonProps): JSX.Element => {
  return (
    <div className="radioButtons">
      <div className="leftRadioButtons">
        <div className="radioButton">
          <button
            onClick={() => {
              props.setAllClicked(true);
              props.setRegisteredClicked(false);
              props.setSeatsClicked(false);
              props.setBookMarkedClicked(false);
            }}>
            <FontAwesomeIcon icon={props.allClicked ? faCircle : faCircleDot} />
          </button>
          <p>ALL</p>
        </div>
        <div className="radioButton">
          <button
            onClick={() => {
              props.setAllClicked(false);
              props.setRegisteredClicked(true);
            }}>
            <FontAwesomeIcon icon={props.registeredClicked ? faCircle : faCircleDot} />
          </button>
          <p>REGISTERED</p>
        </div>
      </div>
      <div className="rightRadioButtons">
        <div className="radioButton">
          <p>BOOKMARKED</p>
          <button
            onClick={() => {
              props.setAllClicked(false);
              props.setBookMarkedClicked(true);
            }}>
            <FontAwesomeIcon icon={props.bookMarkedClicked ? faCircle : faCircleDot} />
          </button>
        </div>
        <div className="radioButton">
          <p>SEATS AVAILABLE</p>
          <button
            onClick={() => {
              props.setAllClicked(false);
              props.setSeatsClicked(true);
            }}>
            <FontAwesomeIcon icon={props.seatsClicked ? faCircle : faCircleDot} />
          </button>
        </div>
      </div>
    </div>
  );
};
export default RadioButtons;
