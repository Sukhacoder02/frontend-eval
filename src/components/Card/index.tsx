import * as React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faBookmark as faBookMarkRegular } from '@fortawesome/free-regular-svg-icons';
import { faCircleCheck, faBookmark as faBookmarkSolid, faCircleXmark } from '@fortawesome/free-solid-svg-icons';

import './Card.css';
import { EventData } from '../../types';
import formatDate from '../../utils/common/dateTimeFormat';
import makeRequest from '../../utils/MakeRequest';
import { PATCH_SINGLE_EVENT } from '../../constants/apiEndPoints';

const Card: React.FC<EventData> = (props: EventData): JSX.Element => {
  const [bookMarked, setBookMarked] = React.useState<boolean>(props.isBookmarked);

  const bookMarkButtonClick = async () => {
    console.log('bookmarked click');
    await makeRequest(PATCH_SINGLE_EVENT(props.id.toString()), {
      data: {
        isBookmarked: !bookMarked,
      },
    });
    setBookMarked(!bookMarked);
  };
  return (
    <div className={'eventCard ' + 'theme' + props.theme}>
      <img src={props.imgUrl} alt="event" />
      <hr />
      <div className="eventTitle">{props.name.toUpperCase()}</div>
      <div className="eventDescription cardPadding">{props.description}</div>
      <div className="eventVenue cardPadding">
        <span>VENUE: </span>
        {props.venue}
      </div>
      <div className="eventDate cardPadding">
        <span>DATE: </span>
        {formatDate(props.datetime)}
      </div>
      <div className="eventButtons cardPadding">
        {props.isRegistered ? (
          <div className="registered">
            <FontAwesomeIcon color="#A0F3AD" icon={faCircleCheck} />
            <span className="green">REGISTERED</span>
          </div>
        ) : !props.areSeatsAvailable ? (
          <div className="noSeats">
            <FontAwesomeIcon color="#ECECAB" icon={faCircleXmark} />
            <span className="yellow">NO SEATS AVAILABLE</span>
          </div>
        ) : (
          <div></div>
        )}
        <button type="button" className="right-icon red" onClick={bookMarkButtonClick}>
          <FontAwesomeIcon color="#ea8282" icon={bookMarked ? faBookMarkRegular : faBookmarkSolid} />
        </button>
      </div>
    </div>
  );
};
export default Card;
