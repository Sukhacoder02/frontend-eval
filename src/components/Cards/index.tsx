import * as React from 'react';
import Card from '../Card';
import './Cards.css';
import makeRequest from '../../utils/MakeRequest';
import { GET_ALL_EVENTS } from '../../constants/apiEndPoints';
import { EventData } from '../../types';
import {
  filterEventsByBookMarked,
  filterEventsByRegistered,
  filterEventsBySeatsAvailable,
} from '../../utils/common/filterEvents';

interface CardsProps {
  // searchKey: string;
  theme: any;
  allClicked: boolean;
  registeredClicked: boolean;
  bookMarkedClicked: boolean;
  seatsClicked: boolean;
}

const Cards: React.FC<CardsProps> = (props: CardsProps): JSX.Element => {
  const [events, setEvents] = React.useState<EventData[]>();
  const [error, setError] = React.useState();

  React.useEffect(() => {
    makeRequest(GET_ALL_EVENTS)
      .then(data => {
        if (props.allClicked) {
          data.sort((a: EventData, b: EventData) => (a.datetime < b.datetime ? -1 : 1));
          setEvents(data);
        } else {
          console.log('filtering');
          let newData = data;
          if (props.registeredClicked) {
            newData = filterEventsByRegistered(data);
          }
          if (props.seatsClicked) {
            newData = filterEventsBySeatsAvailable(newData);
          }
          if (props.bookMarkedClicked) {
            newData = filterEventsByBookMarked(newData);
          }
          newData.sort((a: EventData, b: EventData) => (a.datetime < b.datetime ? -1 : 1));

          setEvents(newData);
        }
      })
      .catch(err => {
        setError(err);
      });
  }, [props.seatsClicked, props.registeredClicked]);
  if (error) {
    return <div>Something went wrong</div>;
  }
  if (events) {
    return (
      <div className="cards">
        {events.map((event: any) => (
          <Card
            theme={props.theme}
            key={event.id}
            id={event.id}
            venue={event.venue}
            name={event.name}
            description={event.description}
            areSeatsAvailable={event.areSeatsAvailable}
            datetime={event.datetime}
            timezone={event.timezone}
            isRegistered={event.isRegistered}
            isBookmarked={event.isBookmarked}
            imgUrl={event.imgUrl}
          />
        ))}
      </div>
    );
  }
  return <div>Loading...</div>;
};
export default Cards;
