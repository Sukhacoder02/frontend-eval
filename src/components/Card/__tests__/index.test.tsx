import * as React from 'react';
import Card from '..';
import { render } from '@testing-library/react';
import { mockEvent } from '../../../__mocks__/components/card';

const card = (
  <Card
    theme={1}
    key={mockEvent.id}
    id={mockEvent.id}
    venue={mockEvent.venue}
    name={mockEvent.name}
    description={mockEvent.description}
    areSeatsAvailable={mockEvent.areSeatsAvailable}
    datetime={mockEvent.datetime}
    timezone={mockEvent.timezone}
    isRegistered={mockEvent.isRegistered}
    isBookmarked={mockEvent.isBookmarked}
    imgUrl={mockEvent.imgUrl}
  />
);
describe('Component Card', () => {
  it('Should render without crashing', () => {
    const { asFragment } = render(card);
    expect(asFragment()).toMatchSnapshot();
  });
});
