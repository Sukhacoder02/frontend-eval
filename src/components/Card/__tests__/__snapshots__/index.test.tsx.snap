// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Card Should render without crashing 1`] = `
<DocumentFragment>
  <div
    class="eventCard theme1"
  >
    <img
      alt="event"
      src="https://i.ibb.co/3zbdvWX/battle-of-bands.jpg"
    />
    <hr />
    <div
      class="eventTitle"
    >
      BATTLE OF THE BANDS
    </div>
    <div
      class="eventDescription cardPadding"
    >
      Get ready for Battle of the Bands, where the hottest up-and-coming rock groups will compete for the ultimate prize. With heart-pumping beats and electrifying performances, you won't want to miss this adrenaline-fueled event. Each band will bring their A-game, leaving everything on the stage for the chance to be crowned champion. Don't miss out on the most unforgettable rock competition of the year!
    </div>
    <div
      class="eventVenue cardPadding"
    >
      <span>
        VENUE: 
      </span>
      All Stars Arena, Las Vegas, NV, USA
    </div>
    <div
      class="eventDate cardPadding"
    >
      <span>
        DATE: 
      </span>
      March 1, 2023
    </div>
    <div
      class="eventButtons cardPadding"
    >
      <div
        class="registered"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-circle-check "
          color="#A0F3AD"
          data-icon="circle-check"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 512c141.4 0 256-114.6 256-256S397.4 0 256 0S0 114.6 0 256S114.6 512 256 512zM369 209L241 337c-9.4 9.4-24.6 9.4-33.9 0l-64-64c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0l47 47L335 175c9.4-9.4 24.6-9.4 33.9 0s9.4 24.6 0 33.9z"
            fill="currentColor"
          />
        </svg>
        <span
          class="green"
        >
          REGISTERED
        </span>
      </div>
      <button
        class="right-icon red"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-bookmark "
          color="#ea8282"
          data-icon="bookmark"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 48V487.7C0 501.1 10.9 512 24.3 512c5 0 9.9-1.5 14-4.4L192 400 345.7 507.6c4.1 2.9 9 4.4 14 4.4c13.4 0 24.3-10.9 24.3-24.3V48c0-26.5-21.5-48-48-48H48C21.5 0 0 21.5 0 48z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
